### PR TITLE
Simplify startup of egressgateway-with-sni-proxy

### PR DESCRIPTION
**Pull request by frankbu**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/tasks/traffic-management/egress/wildcard-egress-hosts/test.sh [hunk 1]
```diff
@@ -61,16 +61,12 @@ snip_setup_egress_gateway_with_sni_proxy_2
...
-snip_setup_egress_gateway_with_sni_proxy_5
-
-_verify_same snip_setup_egress_gateway_with_sni_proxy_6 "$snip_setup_egress_gateway_with_sni_proxy_6_out"
-
-_verify_like snip_setup_egress_gateway_with_sni_proxy_7 "$snip_setup_egress_gateway_with_sni_proxy_7_out"
+snip_setup_egress_gateway_with_sni_proxy_4
 _wait_for_deployment istio-system istio-egressgateway-with-sni-proxy
 
-snip_setup_egress_gateway_with_sni_proxy_8
+_verify_same snip_setup_egress_gateway_with_sni_proxy_5 "$snip_setup_egress_gateway_with_sni_proxy_5_out"
```
Comment (ericvn) [inline content/en/docs/tasks/traffic-management/egress/wildcard-egress-hosts/test.sh:67]: From the test failure, looks like this should be a _verify_like

File: content/en/docs/tasks/traffic-management/egress/wildcard-egress-hosts/index.md [hunk 2]
```diff
@@ -350,43 +350,34 @@ The SNI proxy will forward the traffic to port `443`.
...
+                  securityContext:
+                    runAsNonRoot: false
+                    runAsUser: 0
+              - path: spec.template.spec.volumes[-1]
+                value: |
+                  name: sni-proxy-config
+                  configMap:
+                    name: egress-sni-proxy-configmap
+                    defaultMode: 292 # 0444
     EOF
```
Comment (frankbu) [inline content/en/docs/tasks/traffic-management/egress/wildcard-egress-hosts/index.md:374]:
```suggestion
      values:
        gateways:
          istio-egressgateway:
            runAsRoot: true
    EOF
```

File: content/en/docs/tasks/traffic-management/egress/wildcard-egress-hosts/index.md [hunk 3]
```diff
@@ -330,14 +330,14 @@ The SNI proxy will forward the traffic to port `443`.
 1.  Create an `IstioOperator` CR to add a new egress gateway with SNI proxy:
 
     {{< text bash >}}
-    $ cat > ./egressgateway-with-sni-proxy.yaml <<EOF
+    $ istioctl manifest generate --set values.gateways.istio-egressgateway.runAsRoot=true -f - <<EOF > ./egressgateway-with-sni-proxy.yaml
```
Comment (frankbu) [inline content/en/docs/tasks/traffic-management/egress/wildcard-egress-hosts/index.md:333]:
```suggestion
    $ istioctl manifest generate -f - <<EOF > ./egressgateway-with-sni-proxy.yaml
```

**Review by ericvn — APPROVED** [review]

**Pull request merged**